### PR TITLE
Using temporary files in tests

### PR DESCRIPTION
**Pull request by laserkelvin**
This commit addresses issue [#736], where `test_image_collection.py`
in the test suite generates a `test_table.txt` in the working directory
and does not clean up after itself. This is fixed now by using a
`NamedTemporaryFile` within a context manager, and deletes the file
once the test is completed.

For bugfixes:

- [x] Did you add an entry to the "Changes.rst" file?
- [ ] Did you add a regression test?
- [x] Does the commit message include a "Fixes #issue_number" (replace "issue_number").
- [ ] Does this PR add, rename, move or remove any existing functions or parameters?

-----------------------------------------

### CODE REVIEW

**Comment by mwcraig** [comment]
Can you please add yourself to the contributors list @laserkelvin? It is in `AUTHORS.rst`

**Comment by mwcraig** [comment]
Also, it looks like the Windows test failed...so maybe skip that one on windows like we do the test below it.

File: CHANGES.rst [hunk 1]
```diff
@@ -12,6 +12,8 @@ Other Changes and Additions
 Bug Fixes
 ^^^^^^^^^
 
+- ``test_image_collection.py`` in the test suite no longer produces
+ permanent files on disk and cleans up after itself. [#736]
```
Comment (mwcraig) [inline CHANGES.rst:16]: Please also add the PR number here

Comment (laserkelvin) [reply CHANGES.rst]: I pushed the changes, do I have to do anything to this PR?

**Comment by laserkelvin** [comment]
> Can you please add yourself to the contributors list @laserkelvin? It is in `AUTHORS.rst`

**Comment by mwcraig** [comment]
Did you mean to close this, @laserkelvin?

**Comment by laserkelvin** [comment]
> Did you mean to close this, @laserkelvin?

I did not! My mistake!

I have added my name to `AUTHORS.rst`.

**Comment by mwcraig** [comment]
ok, excellent!

**Comment by mwcraig** [comment]
I'm not quite sure why astropy-bot is complaining about the changelog entry, since it looks like it is in the right place...

**Comment by mwcraig** [comment]
...so will just ignore it

**Comment by mwcraig** [comment]
There is a real failure in the travis run on windows...some sort of permission error that is, I think, travis-specific. @laserkelvin could you please mark the test you are working on as skip on windows? There is an example of how to do that in the test just after the one you are working on.

**Comment by laserkelvin** [comment]
@mwcraig It might also be an issue with how `NamedTemporaryFiles` handles in Windows, as opposed to Linux. I'll see about flagging the test to skip for now, but maybe later on I'll also take a look at seeing if there's a OS-independent solution.

**Comment by mwcraig** [comment]
Looks like this is missing your changes to `AUTHORS.rst`?  Last thing to be done, I think, @laserkelvin

**Comment by laserkelvin** [comment]
> Looks like this is missing your changes to `AUTHORS.rst`? Last thing to be done, I think, @laserkelvin

I'm already on the `AUTHORS.rst` list on `master`, or am I missing something?

**Comment by mwcraig** [comment]
> I'm already on the AUTHORS.rst list on master, or am I missing something?

oops, my bad, merging!

**Pull request merged**